client/src/components/sources/sourceHostList.js

```diff
@@ -1,84 +1,188 @@
 import _ from 'lodash';
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 
-import { EmptyState, Grid, Icon } from 'patternfly-react';
+import { EmptyState, Grid, Icon, Pager } from 'patternfly-react';
 import helpers from '../../common/helpers';
+import { getConnectionScanResults } from '../../redux/actions/scansActions';
 
-const SourceHostList = ({ source, scanResults, scanResultsError, scanResultsPending, status }) => {
-  if (scanResultsPending === true) {
-    return (
-      <EmptyState>
-        <EmptyState.Icon name="spinner spinner-xl" />
-        <EmptyState.Title>Loading scan results...</EmptyState.Title>
-      </EmptyState>
-    );
+class SourceHostList extends React.Component {
+  constructor() {
+    super();
+
+    helpers.bindMethods(this, ['onNextPage', 'onPreviousPage']);
+
+    this.state = {
+      scanResults: [],
+      scanResultsError: false,
+      scanResultsPending: false,
+      page: 1,
+      disableNext: true,
+      disablePrevious: true
+    };
   }
 
-  if (scanResultsError) {
-    return (
-      <EmptyState>
-        <EmptyState.Icon name="error-circle-o" />
-        <EmptyState.Title>Error retrieving scan results</EmptyState.Title>
-        <EmptyState.Info>{scanResultsError}</EmptyState.Info>
-      </EmptyState>
-    );
+  componentDidMount() {
+    this.refresh();
   }
 
-  let results = _.get(scanResults, 'results', []);
-  let displayHosts = [];
-  _.forEach(results, result => {
-    if (result.status === status && _.get(result, 'source.id') === source.id) {
-      displayHosts.push(result);
+  componentWillReceiveProps(nextProps) {
+    // Check for changes resulting in a fetch
+    if (!_.isEqual(nextProps.lastRefresh, this.props.lastRefresh)) {
+      this.refresh();
     }
-  });
+  }
+
+  onNextPage() {
+    this.setState({ page: this.state.page + 1 });
+    this.refresh(this.state.page + 1);
+  }
+
+  onPreviousPage() {
+    this.setState({ page: this.state.page - 1 });
+    this.refresh(this.state.page - 1);
+  }
+
+  refresh(page) {
+    const { source } = this.props;
+
+    this.setState({
+      scanResultsPending: true,
+      disableNext: true,
+      disablePrevious: true
+    });
+
+    const queryObject = {
+      page: page === undefined ? this.state.page : page,
+      page_size: 100,
+      ordering: 'name'
+    };
+
+    if (_.get(source, 'connection.id')) {
+      this.props
+        .getConnectionScanResults(source.connection.id, queryObject)
+        .then(results => {
+          console.log('ConnectionScanResults');
+          console.dir(results);
+          console.log(_.get(results.value, 'data.next'));
```

Comment:
There's a few of these hanging around... see also `client/src/components/sources/sourceListItem.js`

Reply:
Oops. Removed. Thanks.